code/modules/mob/living/blood.dm

```diff
@@ -36,50 +37,56 @@
 		if(blood_volume < BLOOD_VOLUME_NORMAL)
 			blood_volume += 0.1 // regenerate blood VERY slowly
 
-		var/b_volume = blood_volume
-
 		// Damaged heart virtually reduces the blood volume, as the blood isn't
 		// being pumped properly anymore.
 		if(species && species.has_organ["heart"])
 			var/datum/internal_organ/heart/heart = internal_organs_by_name["heart"]
 
 			if(!heart)
-				b_volume = 0
-			else if(reagents.get_reagent_amount("peridaxon") >= 0.05)
-				b_volume *= 1
-			else if(heart.damage > 1 && heart.damage < heart.min_bruised_damage)
-				b_volume *= 0.8
-			else if(heart.damage >= heart.min_bruised_damage && heart.damage < heart.min_broken_damage)
-				b_volume *= 0.6
-			else if(heart.damage >= heart.min_broken_damage && heart.damage < INFINITY)
-				b_volume *= 0.3
+				heart_multi += 1 //you'd die in seconds but you can't remove internal organs even with varediting.
+
+			if((!reagents.get_reagent_amount("peridaxon") >= 0.05) && heart.damage > 1)
+				if(heart.damage < heart.min_bruised_damage)
+					heart_multi = 1.2
+					blood_volume -= 0.1 //nulls regeneration
+				else if(heart.damage >= heart.min_bruised_damage && heart.damage < heart.min_broken_damage)
+					heart_multi = 1.5
+					blood_volume -= 0.5
+				else if(heart.damage >= heart.min_broken_damage && heart.damage < INFINITY)
+					heart_multi = 1.8
+					blood_volume -= 1.3
+			else
+				heart_multi = 1
 
 
 
 
 
 	//Effects of bloodloss
-		switch(blood_volume)
+		switch(blood_volume / heart_multi)
```

Comment:
Can you change the `heart_multi` logic to avoid a division here? Multiplying `blood_volume` by 0.5 would achieve the same result than dividing by 2, for example.